Assets/CarriageManager.cs

```diff
@@ -0,0 +1,143 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CarriageManager : MonoBehaviour
+{
+    public List<GameObject> carriagePrefabs;
+
+    private List<GameObject> carriagesToSpawn;
+    [SerializeField] private List<GameObject> spawnedCarriages;
+    [SerializeField] private CarriageData currentCarriage, oldCarriage;
+
+    //SPAWNING
+    [SerializeField] private Transform firstSpawn;
+    [SerializeField] private Transform nextSpawnPos,prevSpawnPos;
+
+    //MOVING
+    private void Awake()
+    {
+    }
+    private void Start()
+    {
+        InitialiseCarriages();
+    }
+
+    private void InitialiseCarriages()
+    {
+        //ALGORITHM CAN BE ALTERED TO BE RANDOMISED ON EVERY NEW PLAYTHROUGH OR KEPT AS IS
+
+        carriagesToSpawn = new List<GameObject>(carriagePrefabs.Count);
+        foreach (GameObject _carriage in carriagePrefabs)
```

Comment:
maybe get into the habit of using `{` `}` even when it is optional? It improves readability of code and makes it quicker to come in later on and add another statement into the `foreach` loop if you needed to